tests/license_header_check.py

```diff
@@ -0,0 +1,105 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import sys
+from pathlib import Path
+
+paths = ["rust/src", "tests"]
+
+headers = [
+    "#\\s|\\SSPDX-License-Identifier: LGPL-2.1-or-later",
+    "//\\s|\\SSPDX-License-Identifier: Apache-2.0",
```

Comment:
The regex are wrong.
```
regex = "#\\s|\\SSPDX-License-Identifier: LGPL-2.1-or-later"

# this will match
line = "# "
re.match(regex, line)

# this too
line = "ASPDX-License-Identifier: LGPL-2.1-or-later"
re.match(regex, line)
```

You need to remove the `|\\S` from both of them.